policy/policy-authority-data.go

```diff
@@ -40,20 +43,27 @@ type RuleSet struct {
 	Whitelist []WhitelistRule
 }
 
+type gorpDbMap interface {
+	AddTableWithName(interface{}, string) *gorp.TableMap
+	Begin() (*gorp.Transaction, error)
+	SelectOne(interface{}, string, ...interface{}) error
+	Select(interface{}, string, ...interface{}) ([]interface{}, error)
+}
+
 // PolicyAuthorityDatabaseImpl enforces policy decisions based on various rule
 // lists
 type PolicyAuthorityDatabaseImpl struct {
 	log   *blog.AuditLogger
-	dbMap *gorp.DbMap
+	dbMap gorpDbMap
 }
 
 // NewPolicyAuthorityDatabaseImpl constructs a Policy Authority Database (and
 // creates tables if they are non-existent)
-func NewPolicyAuthorityDatabaseImpl(dbMap *gorp.DbMap) (padb *PolicyAuthorityDatabaseImpl, err error) {
+func NewPolicyAuthorityDatabaseImpl(dbMap gorpDbMap) (padb *PolicyAuthorityDatabaseImpl, err error) {
 	logger := blog.GetAuditLogger()
 
-	dbMap.AddTableWithName(BlacklistRule{}, "blacklist").SetKeys(false, "Host")
-	dbMap.AddTableWithName(WhitelistRule{}, "whitelist").SetKeys(false, "Host")
+	dbMap.AddTableWithName(BlacklistRule{}, "blacklist")
+	dbMap.AddTableWithName(WhitelistRule{}, "whitelist")
```

Comment:
SetKeys is used when you're asking Gorp to create the table for you. In our case we're not: we use migrations instead.